src/pages/checkout.tsx

```diff
@@ -1,13 +1,111 @@
+                    <div className="row">
+                        <div className="col-lg-8 col-md-6" >
+                            <PromoCodeWithLink hasRef />
+                            <CheckoutForm isSubmited={submitedCount} handleFormSubmit={handleFormSubmit} />
+                        </div>
+                        <div className="col-lg-4 col-md-6">
+                            <div className="checkout__order">
+                                <h4 className="order__title">Your order</h4>
+                                <div className="checkout-order__products">
+                                    Product <span>Total</span>
+                                </div>
+
+                                {basketItems &&
+                                    <ul className="checkout-order__total-products">
+                                        {basketItems.map((p, index) => {
+                                            const formattedNumber = (index + 1).toLocaleString('en-US', {
+                                                minimumIntegerDigits: 2,
+                                                useGrouping: false
+                                            })
+                                            return <>
+                                                <li key={"product-ordered-" + p.id}>
+                                                    <span>{formattedNumber}.</span>
+                                                    <span>{p.title} <strong>({p.quantity})</strong></span>
+                                                    <span>$ {(p.quantity * p.price).toFixed(2)}</span>
+                                                </li>
+                                            </>
+                                        })}
+
+                                    </ul>
+                                }
+                                <div className="checkout-order__total-all"><CartTotal /></div>
+                                <p>
+                                    Lorem ipsum dolor sit amet, consectetur adip elit, sed do
+                                    eiusmod tempor incididunt ut labore et dolore magna aliqua.
+                                </p>
+                                <PaymentMethodForm isSubmited={submitedCount} handleFormSubmit={handlePaymentFormSubmit} />
+
+
+                                {paymentMethod !== PAYMENT_METHOD.PAYPAL
+                                    && <BlackButton type="submit" cssClass='site-btn' handleClick={handleCheckout}>
+                                        <>PLACE ORDER</>
+                                    </BlackButton>
+                                }
+
+                                {paymentMethod === PAYMENT_METHOD.PAYPAL
+                                    && Object.values(checkoutInfo).length > 0
+                                    && <PaypalButton />}
+
+                                {paymentMethod === PAYMENT_METHOD.PAYPAL
```

Comment:
too many logic here, do you consider breaking it out into function render?

for example `renderBlackButton()`, `renderPaypalButton()`